src/main/java/net/comments/ft/Test2.java

```diff
@@ -1,46 +1,46 @@
 package net.comments.ft;
 
+import net.comments.objects.*;
+import net.comments.selenium.CommentsDriver;
 import org.hamcrest.MatcherAssert;
-import org.hamcrest.Matchers;
-import org.openqa.selenium.By;
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
-import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
-
 public class Test2 {
-    private final WebDriver[] driver = new WebDriver[1];
+    private final CommentsDriver driver = new CommentsDriver();
+    private final CommentsPage commentsPage;
+    private final CommentsTable commentsTable;
+    private final CommentsActions commentsActions;
+    private final ModifyCommentPage modifyComment;
+    private final ModifyCommentActions modifyCommentAction;
+
+    public Test2() {
+        this.commentsPage = new BCommentsPage(driver);
+        this.commentsTable = new BCommentsTable(driver);
+        this.commentsActions = new BCommentsActions(driver);
+        this.modifyComment = new BModifyCommentPage(driver);
+        this.modifyCommentAction = new BModifyCommentActions(driver);
+    }
 
     @Test
     public void test() {
-        this.driver().get("http://commentssprintone.azurewebsites.net");
-        this.driver().findElements(By.name("SelectedId")).get(3).click();
-        this.driver().findElement(By.xpath("//*[@id=\"command-navigation\"]/input[1]")).click();
-        this.driver().findElement(By.xpath("//*[@id=\"Text\"]")).clear();
-        this.driver().findElement(By.xpath("//*[@id=\"Text\"]")).sendKeys("New Duplicated Comment");
-        this.driver().findElement(By.xpath("//*[@id=\"Number\"]")).clear();
-        this.driver().findElement(By.xpath("//*[@id=\"Number\"]")).sendKeys("399");
-        this.driver().findElement(By.xpath("//*[@id=\"editor-navigation\"]/input[2]")).click();
-        this.driver().findElement(By.xpath("//*[@id=\"main\"]/div/div[5]/form/table/tfoot/tr/td/a[3]")).click();
-        final String page = this.driver().getPageSource();
-        MatcherAssert.assertThat("New duplicated comment text is not shown", page.contains("New Duplicated Comment"));
-        MatcherAssert.assertThat(this.driver().findElement(By.xpath("//*[@id=\"main\"]/div/div[5]/form/table/tbody/tr/td[2]")).getText(), Matchers.is("399"));
-
+        commentsPage.open();
+        commentsTable.selectComment("5");
+        commentsActions.duplicate();
+        String commentNumber = "555";
+        modifyComment.fillNumber(commentNumber);
+        modifyCommentAction.saveAndReturn();
+        commentsPage.commentsFrom(4);
```

Comment:
It will be great if `commentsFrom` has a more descriptive name. For me, the line says: "Give me comments stating from 4". But the proper meaning is "Open fourth page with comments".